sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -0,0 +1,958 @@
+         Default value is None.
+        :paramtype temperature: float
+        :keyword top_p: An alternative to sampling with temperature called nucleus sampling. This value
+         causes the
+         model to consider the results of tokens with the provided probability mass. As an example, a
+         value of 0.15 will cause only the tokens comprising the top 15% of probability mass to be
+         considered.
+         It is not recommended to modify temperature and top_p for the same completions request as the
+         interaction of these two settings is difficult to predict.
+         Supported range is [0, 1].
+         Default value is None.
+        :paramtype top_p: float
+        :keyword max_tokens: The maximum number of tokens to generate. Default value is None.
+        :paramtype max_tokens: int
+        :keyword response_format: An object specifying the format that the model must output. Used to
+         enable JSON mode. Known values are: "text" and "json_object". Default value is None.
+        :paramtype response_format: str or ~azure.ai.inference.models.ChatCompletionsResponseFormat
+        :keyword stop: A collection of textual sequences that will end completions generation. Default
+         value is None.
+        :paramtype stop: list[str]
+        :keyword stream: A value indicating whether chat completions should be streamed for this request.
+         Default value is False. If streaming is enabled, the response will be a StreamingChatCompletions.
+         Otherwise the response will be a ChatCompletions.
+        :paramtype stream: bool
+        :keyword tools: The available tool definitions that the chat completions request can use,
+         including caller-defined functions. Default value is None.
+        :paramtype tools: list[~azure.ai.inference.models.ChatCompletionsToolDefinition]
+        :keyword tool_choice: If specified, the model will configure which of the provided tools it can
+         use for the chat completions response. Is either a Union[str,
+         "_models.ChatCompletionsToolSelectionPreset"] type or a ChatCompletionsNamedToolSelection type.
+         Default value is None.
+        :paramtype tool_choice: str or ~azure.ai.inference.models.ChatCompletionsToolSelectionPreset or
+         ~azure.ai.inference.models.ChatCompletionsNamedToolSelection
+        :keyword seed: If specified, the system will make a best effort to sample deterministically
+         such that repeated requests with the
+         same seed and parameters should return the same result. Determinism is not guaranteed.".
+         Default value is None.
+        :paramtype seed: int
+        :return: ChatCompletions for non-streaming, or StreamingChatCompletions for streaming.
+        :rtype: ~azure.ai.inference.models.ChatCompletions or ~azure.ai.inference.models.StreamingChatCompletions
+        :raises ~azure.core.exceptions.HttpResponseError
+        """
+        error_map = {
```

Comment:
I think we can replace this method body with:

```python
        if body is _Unset:
            body = {
                "frequency_penalty": frequency_penalty,
                "max_tokens": max_tokens,
                "messages": messages,
                "presence_penalty": presence_penalty,
                "response_format": response_format,
                "seed": seed,
                "stop": stop,
                "stream": stream,
                "temperature": temperature,
                "tool_choice": tool_choice,
                "tools": tools,
                "top_p": top_p,
            }
            body.update(model_extras or {})
            return super().complete({k:v for k, v in body.items() if v is not None})
        return super().complete(
            body,
            messages= messages,
            model_extras = model_extras,
            frequency_penalty = frequency_penalty,
            presence_penalty = presence_penalty,
            temperature = temperature,
            top_p = top_p,
            max_tokens = max_tokens,
            response_format = response_format,
            stop = stop,
            stream = stream,
            tools = tools,
            tool_choice = tool_choice,
            seed = seed,
            **kwargs,       
        )
```

(I'm probably missing some mapping)

Reply:
Yeah, missed the defaulted header for dealing with extra parameters. But still simpled than copying the whole method body, no?

Reply:
Unfortunately, we can't do that, as I need special code to deal with the streaming case. For streaming, the return object should be `_models.StreamingChatCompletions(response)` which is not provided by `super().complete()`.